torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2527,8 +2527,15 @@ def _sharded_pre_load_state_dict_hook(
             param = state_dict.pop(fqn)
 
             # All-gather the param (ShardedTensor)
-            shards = param.local_shards()
-            local_tensor = cast(torch.Tensor, shards[0].tensor).flatten()
+            param, shards = _ext_pre_load_state_dict_transform(param)
+            assert len(shards) < 2, (
+                f"Expects 0 or 1 shard per rank but got {len(shards)} shards on rank {self.rank}"
+            )
+            if shards:
+                local_tensor = shards[0].flatten()
+            else:
+                # May have no shards, e.g. shape (1, 1, 756)
+                local_tensor = torch.tensor([], dtype=param.dtype, device=self.compute_device)
```

Comment:
Just realized I already added some checks in https://github.com/pytorch/pytorch/pull/85640.

Reply:
I rebased and handled the rebase conflict. Feel free to take a second look to see if it is what you expect.